src/block_disagg/block_disagg_read.c

```diff
@@ -86,6 +107,7 @@ __block_disagg_read_multiple(WT_SESSION_IMPL *session, WT_BLOCK_DISAGG *block_di
     WT_STAT_CONN_INCR(session, disagg_block_get);
     WT_STAT_CONN_INCR(session, block_read);
     WT_STAT_CONN_INCRV(session, block_byte_read, size);
+    __block_disagg_check_lsn_frontier(session, lsn);
```

Comment:
Not related to your changes but it feels weird to me that we update `block_read` and `block_byte_read`, same for the `hs` stats below before proceeding with the actual read.

Reply:
That could be represented as a counter for read attempts, not successful reads.